api/class-wc-rest-dev-mailchimp-settings-controller.php

```diff
@@ -0,0 +1,307 @@
+	 *
+	 */
+	public function update_newsletter_settings( $request ) {
+		$parameters                         = $request->get_params();
+		$parameters['mailchimp_active_tab'] = 'newsletter_settings';
+		$handler                            = MailChimp_Woocommerce_Admin::connect();
+		$options                            = get_option( 'mailchimp-woocommerce', array() );
+		$mailchimp_active_tab               = $options['active_tab'];
+	  $data                               = $handler->validate( $parameters );
+
+		// if previous active tab was sync then we still want sync
+		// because this call is just an update and not part of setup
+		if( 'sync' === $mailchimp_active_tab  ) {
+			$data['mailchimp_active_tab'] = 'sync';
+		}
+
+		update_option( 'mailchimp-woocommerce', $data);
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Synchronization status between MailChimp plugin and Mailchimp
+	 *
+	 * @param WP_REST_Request,
+	 * @return WP_REST_Response, synchronization status
+	 *
+	 */
+	public function get_sync_status( $request ) {
+		$handler                  = MailChimp_Woocommerce_Admin::connect();
+		$mailchimp_total_products = $mailchimp_total_orders = 0;
+		$store_id                 = mailchimp_get_store_id();
+		$product_count            = mailchimp_get_product_count();
+		$order_count              = mailchimp_get_order_count();
+		$store_syncing            = false;
+		$last_updated_time        = get_option( 'mailchimp-woocommerce-resource-last-updated' );
+		$account_name             = 'n/a';
+		$mailchimp_list_name      = 'n/a';
+
+		if (!empty($last_updated_time)) {
+		    $last_updated_time = mailchimp_date_local( $last_updated_time );
+		}
+
+		if ( ( $mailchimp_api = mailchimp_get_api() ) && ( $store = $mailchimp_api->getStore( $store_id ) ) ) {
+
+		    $store_syncing = $store->isSyncing();
+
+		    if ( ( $account_details = $handler->getAccountDetails() ) ) {
+		        $account_name = $account_details['account_name'];
+		    }
+
+		    try {
+		        $products = $mailchimp_api->products( $store_id, 1, 1 );
+		        $mailchimp_total_products = $products['total_items'];
+		        if ( $mailchimp_total_products > $product_count ) $mailchimp_total_products = $product_count;
+		    } catch (\Exception $e) { $mailchimp_total_products = 0; }
+
+		    try {
+		        $orders = $mailchimp_api->orders( $store_id, 1, 1 );
+		        $mailchimp_total_orders = $orders['total_items'];
+		        if ( $mailchimp_total_orders > $order_count ) $mailchimp_total_orders = $order_count;
+		    } catch (\Exception $e) { $mailchimp_total_orders = 0; }
+
+		    $mailchimp_list_name = $handler->getListName();
+		}
+
+		$data = array();
+
+		$data['last_updated_time']        = $last_updated_time->format( 'D, M j, Y g:i A' );
+		$data['store_syncing']            = $store_syncing;
+		$data['mailchimp_total_products'] = $mailchimp_total_products;
+		$data['product_count']            = $product_count;
```

Comment:
It's still not totally clear to me what the difference between the counts here and the mailchimp total counts. Both seem to be coming from mailchimp data/functions.

Can you add a comment about them here?